### ISSUE TITLE
Bi-weave regen numbers seem to be off (video evidence)

### ISSUE THREAD

**Issue by alterNERDtive** [EDCD/coriolis#589]
C&P from Discord; I’m planning to do an actual timed recording once the account is back in the bubble, probably end of the week or something.

-----

so something about bi-weaves regen is not quite right. i’ve tested this build https://s.orbis.zone/6_69; if you look at this https://www.twitch.tv/videos/629878204?t=2h59m40s, i’m losing shields, and about 45s later they are up again. that’s ~16s until it starts regenerating, and then ~30s until back up. with me messing up pips momentarily, even. now even if the 55s coriolis gives include the time until the shield has rebooted and starts regenerating, it’s still off. or i’m missing something which is also quite possible :)

also regen is way more than the ~.5%/s the stats suggest, see https://www.twitch.tv/videos/629878204?t=2h54m35s (almost 1%/s which would be a recharge time of ~50s, no? not 67s)

**Comment by alterNERDtive** [comment]
Build link is taken straight from EDSM’s fleet page btw, so should be (mostly? :D) accurate to what it is ingame.

**Comment by alex-williams** [comment]
Without further evidence, I believe Coriolis calculation to be accurate. Obviously, Coriolis calculations are 'ideal world', where the pips are already set from the point where the shield fails, it will never quite match real world scenarios, where pips are being shifted around in a combat scenario. Regen in Coriolis, simply means 0% shield to 50% with 4 pips in Sys, Recharge is 50% to 100%, with 4 pips in sys. It's a good indication, even if not perfectly accurate to a given combat scenario.

**Comment by alterNERDtive** [comment]
According to you, Coriolis should be too optimistic. It’s the opposite case, here.

Corioslis says 0–50% should be 43s with 4 pips in SYS (this value actually changes with pip assignment on the site), and 50%–100% should be 1m07s. (these are the current numbers, obviously; I’m not going to run a 2020 version locally to get the ones from back then)

Elite (well, 2020 Elite; I cannot test current) had 0–50% at ~16s and 50%–100% at ~30s; without constant 4 pips in SYS. This is _significantly_ better than what Coriolis reports, at ~37% and ~45% of the values, respectively.

If Coriolis were an “ideal world” representation, then it should _never_ report numbers that are too high.

(also yay Twitch, gj deleting old vods)

**Comment by alex-williams** [comment]
> According to you, Coriolis should be too optimistic. It’s the opposite case, here.
> 
> Corioslis says 0–50% should be 43s with 4 pips in SYS (this value actually changes with pip assignment on the site), and 50%–100% should be 1m07s. (these are the current numbers, obviously; I’m not going to run a 2020 version locally to get the ones from back then)
> 
> Elite (well, 2020 Elite; I cannot test current) had 0–50% at ~16s and 50%–100% at ~30s; without constant 4 pips in SYS. This is _significantly_ better than what Coriolis reports, at ~37% and ~45% of the values, respectively.
> 
> If Coriolis were an “ideal world” representation, then it should _never_ report numbers that are too high.
> 
> (also yay Twitch, gj deleting old vods)

Obviously, we don't need to look at 2020, but without fresh evidence that the calculations are off, I have to assume they're correct. I unfortunately do not get the time to play very much at all now, so it's not something I can test for myself. If you want to provide some testing to prove the point that the calculations are off, I'm more than happy to review that testing and try to update the predictions. I myself also thought for years that the recharge and recover rates weren't accurate, before I became the maintainer, but looking at the code, it makes sense.
